histomicstk/FeatureExtraction.py

```diff
@@ -129,6 +177,410 @@ def FeatureExtraction(Label, I, K=128, Fs=6, Delta=8):
     df['Extent'] = Extent
     df['Solidity'] = Solidity
 
-    # df.insert(0, 'Slide', SlideName)
+    for i in range(0, Fs):
+        df['FSD' + str(i+1)] = FSDGroup[:, i]
+
+    for f in HematoxylinIntensityGroup._fields:
+        df['Hematoxylin' + f] = getattr(HematoxylinIntensityGroup, f)
+
+    for f in HematoxylinTextureGroup._fields:
+        df['Hematoxylin' + f] = getattr(HematoxylinTextureGroup, f)
+
+    for f in HematoxylinGradientGroup._fields:
+        df['Hematoxylin' + f] = getattr(HematoxylinGradientGroup, f)
+
+    for f in EosinIntensityGroup._fields:
+        df['Cytoplasm' + f] = getattr(EosinIntensityGroup, f)
+
+    for f in EosinTextureGroup._fields:
+        df['Cytoplasm' + f] = getattr(EosinTextureGroup, f)
+
+    for f in EosinGradientGroup._fields:
+        df['Cytoplasm' + f] = getattr(EosinGradientGroup, f)
 
     return df
+
+
+def GradientFeatureGroup(I, Coords):
+    """
+    Get GradientFeatures for nuclei and cytoplasms
+    Parameters
+    ----------
+    I : array_like
+        A T x T intensity image.
+    Coords : array_like
+        A N x 2 coordinate list of a region.
+    Returns
+    -------
+    MeanGradMag : float
+        Mean of gradient data.
+    StdGradMag : float
+        Standard deviation of gradient data.
+    EntropyGradMag : float
+        Entroy of gradient data.
+    EnergyGradMag : float
+        Energy of gradient data.
+    SkewnessGradMag : float
+        Skewness of gradient data. Value is 0 when all values are equal.
+    KurtosisGradMag : float
+        Kurtosis of gradient data. Value is -3 when all values are equal.
+    SumCanny : float
+        Sum of canny filtered gradient data.
+    MeanCanny : float
+        Mean of canny filtered gradient data.
+    Notes
+    -----
+    Return values are returned as a namedtuple.
```

Comment:
Hi Deepak,

I really appreciate your valuable comments all. Could you let me know what
the components of the tuple are? Does that mean types of values (e.g. float
or array-like)?

best,


Sanghoon Lee


2016-05-09 11:36 GMT-04:00 Deepak Roy Chittajallu <notifications@github.com>
:

> In histomicstk/FeatureExtraction.py
> <https://github.com/DigitalSlideArchive/HistomicsTK/pull/82#discussion_r62520371>
> :
>
> > +        Standard deviation of gradient data.
> > +    EntropyGradMag : float
> > +        Entroy of gradient data.
> > +    EnergyGradMag : float
> > +        Energy of gradient data.
> > +    SkewnessGradMag : float
> > +        Skewness of gradient data. Value is 0 when all values are equal.
> > +    KurtosisGradMag : float
> > +        Kurtosis of gradient data. Value is -3 when all values are equal.
> > +    SumCanny : float
> > +        Sum of canny filtered gradient data.
> > +    MeanCanny : float
> > +        Mean of canny filtered gradient data.
> > +    Notes
> > +    -----
> > +    Return values are returned as a namedtuple.
>
> @slee172 <https://github.com/slee172> For all places where you return a
> named tuple, can you please add the docstring for the components of the
> tuple?
>
> Here
> <https://github.com/DigitalSlideArchive/HistomicsTK/blob/master/histomicstk/ColorDeconvolution.py#L30-L41>
> is an example.
>
> Also merge changes from the master repo using git merge origin master and
> resolve the conflicts. Currently it says that the code in your branch has
> some conflict with the code in the master repo. This could be because both
> modified the same file. You will have to go into the conflicted files and
> merge (it will show both your code and the code in the master for each
> conflict. You will have to select the right one and remove the wrong one)
> the changes one by one.
>
> Rest all looks good ! Great job !
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/DigitalSlideArchive/HistomicsTK/pull/82/files/645e9d6a4bef31a864cdc8fd23522b70a6753bef#r62520371>
>